### PR TITLE
refactor discussion groups to ndb

### PR DESCRIPTION
**Pull request by lucasvanhalst**
not tested yet

Migration:

```python
def migrate_discussion_groups():
    from solutions.common.migrations._008_discussion_groups import migrate
    return migrate(dry_run=True)

### CODE REVIEW

File: src/solutions/common/models/discussion_groups.py [hunk 1]
```diff
@@ -42,8 +44,36 @@ def create_key(cls, service_user, discussion_group_id):
...
         if order_by:
-            qry.order(order_by)
+            qry = qry.order(order_by)
         return qry
+
+
+class DiscussionGroup(NdbModel):
+    topic = ndb.StringProperty()
+    description = ndb.TextProperty()
+    members = ndb.StringProperty(repeated=True)  # type: List[str]
```
Comment (RubenMattan) [inline src/solutions/common/models/discussion_groups.py:57]: let's make this more scalable

Comment (lucasvanhalst) [reply src/solutions/common/models/discussion_groups.py:57]: I think we can postpone this, I doubt cities would want to have a chatgroup with 10k+ members.

File: src/solutions/common/models/discussion_groups.py [hunk 2]
```diff
@@ -42,8 +44,36 @@ def create_key(cls, service_user, discussion_group_id):
...
+    def _create_parent_key(service_user):
+        return parent_ndb_key(service_user, SOLUTION_COMMON)
+
+    @classmethod
+    def create_key(cls, service_user, discussion_group_id):
+        return ndb.Key(cls, discussion_group_id, parent=cls._create_parent_key(service_user))
+
+    @classmethod
+    def list(cls, service_user):
+        return cls.query().ancestor(cls._create_parent_key(service_user))
```
Comment (RubenMattan) [inline src/solutions/common/models/discussion_groups.py:75]: ancestor wrong

File: src/solutions/common/models/discussion_groups.py [hunk 3]
```diff
@@ -42,8 +44,36 @@ def create_key(cls, service_user, discussion_group_id):
         return db.Key.from_path(cls.kind(), discussion_group_id, parent=cls._create_parent_key(service_user))
 
     @classmethod
-    def list(cls, service_user, order_by=None, keys_only=False):
-        qry = cls.all(keys_only=keys_only).ancestor(cls._create_parent_key(service_user))
+    def list(cls, service_user, order_by=None):
+        qry = cls.all(ancestor=cls._create_parent_key(service_user))
```
Comment (RubenMattan) [inline src/solutions/common/models/discussion_groups.py:48]: ancestor wrong

File: src/solutions/common/models/discussion_groups.py (hunk 2 above)
Comment (lucasvanhalst) [reply src/solutions/common/models/discussion_groups.py:75]: should be `cls.query(ancestor=...`

File: src/solutions/common/bizz/discussion_groups.py [hunk 4]
```diff
@@ -176,10 +167,10 @@ def poke_discussion_groups(service_user, email, tag, result_key, context, servic
...
     widget = MultiSelectTO()
-    widget.choices = list()
-    widget.values = list()
+    widget.choices = []
+    widget.values = []
     app_user_email = user_details[0].toAppUser().email()
-    for discussion_group in SolutionDiscussionGroup.list(service_user):
+    for discussion_group in DiscussionGroup.list(service_user):
         widget.choices.append(ChoiceTO(label=discussion_group.topic, value=unicode(discussion_group.id)))
         if app_user_email in discussion_group.members['members']:
```
Comment (RubenMattan) [inline src/solutions/common/bizz/discussion_groups.py:175]: not migrated

**Pull request closed without merging**